webdriver/tests/bidi/browsing_context/set_viewport/set_viewport.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+
+from tests.support.image import png_dimensions
+
+from . import layout_viewport_dimensions
+
+@pytest.mark.asyncio
+async def test_capture(bidi_session, top_context, inline):
```

Comment:
Could you please split this test into different scenarios?

* Only setting the width
* Only setting the height
* Setting the width and height
* Resetting the view port to it's original size
* setting the viewport only affects the current tab
* viewport settings persist after a reload / navigation (with and without browsing context replacement - COOP navigation)

For the first 3 tests we could use parameterization.

Also I think that each of the tests should as best run in it's own tab so that it's all cleaned up correctly after the test has been run.

Reply:
Oh, and maybe you could also add a new test under capture screenshot which tests that as well? Reading the proposed test name triggered such a test for me.

Reply:
I think we do not support setting only the width or only the height, currently?  I wonder if we need to perhaps add it to the spec. The behaviour in Chrome is that zero value means the original size.

Reply:
To open a new tab for each test you can use the `new_tab` fixture. Quite a lot of wdspec tests for BiDi make use of it.

Reply:
Yes have a look at the following classic test which exercises all the variants. I assume we only need a normal and coop navigation: https://github.com/web-platform-tests/wpt/blob/master/webdriver/tests/classic/navigate_to/navigate.py#L42-L53.

Reload is not implemented in Firefox yet but @thiagowfx is currently working on adding more reload tests. See https://github.com/web-platform-tests/wpt/pull/40134 for example. So I assume that for Chrome it should work? If not we could add a todo item and add it later.

Reply:
Reload seems to work with chromium-bidi 👍